enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  # my each
+  def my_each
+    return to_enum unless block_given?
+    size.times do |n|
+      yield to_a[n]
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+    n = 0
+    while n < to_a.length
+      yield(to_a[n], n)
+      n += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+    selected = is_a?(Array || Range) ? [] : {}
+    if is_a?(Array)
+      my_each do |i|
+        selected.push(i) if yield(i)
+      end
+    else
+      my_each do |key, val|
+        selected[key] = val if yield(key, val)
+      end
+    end
+    selected
+  end
+
+  # my_all?
+  def my_all?(pattern = nil)
+    result = true
+    my_each do |i|
+      block_given? && pattern.nil? ? result = yield(i) : next 
+      case pattern
+      when nil
+          result = false if !i
+          
+        when Regexp
+        result = false unless i.match(pattern)
+      when Class
+        result = false unless i.is_a?(pattern)
+      when String, Numeric
+        result = false unless i == pattern
+      when nil 
+        result = false unless yield(i)
+      
+      end
+      break if result != true
+    end
+    result
+  end
```

Comment:
#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil if the mentioned condition is not met it should return false

     > this [nil, false, true, []].my_all? should be equal to this [nil, false, true, []].all? => false

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

     > [1, 2.5, 'a', 9].my_all?(Integer) should be equal to this [1, 2.5, 'a', 9].all?(Integer) => false

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

     >  %w[dog door rod blade].my_all?(/d/) should be equal to this %w[dog door rod blade].all?(/d/) => true

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

     > [3,4,7,11].my_all?(3) should be equal to this [3,4,7,11].all?(3)  => false

Reply:
This has been bothering me this whole project. When I set the default result of my_all to true with the conditional of result = false unless i >> nil evaluates to false and it works properly. 
![image](https://user-images.githubusercontent.com/55356496/76068860-3bb80100-5f92-11ea-8eba-87c5ccb7df15.png)
However, if I set the default to false with conditional of result = true if i >> nil evaluates to true and it's not working properly.
![image](https://user-images.githubusercontent.com/55356496/76069026-876aaa80-5f92-11ea-94cc-e51551bd01f0.png)